python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/datahub_client.py

```diff
@@ -191,105 +182,159 @@ def create_domain(
+                ),
+                domain=domain,
+                governance=Governance(
+                    data_owner=owner,
+                    data_stewards=[
+                        OwnerRef(
+                            display_name="", email="Contact email for the user", id=""
+                        )
+                    ],
+                ),
+                tags=tags,
+                platform=EntityRef(display_name=platform_name, id=platform_name),
+                custom_properties=custom_properties,
             )
-            self.graph.emit(metadata_event)
-            logger.info(f"Database {name} associated with domain {domain}")
 
-        database_properties = ContainerPropertiesClass(
-            customProperties={key: str(val) for key, val in metadata_dict.items()},
-            description=description,
-            name=name,
-        )
-        metadata_event = MetadataChangeProposalWrapper(
-            entityType="container",
-            changeType=ChangeTypeClass.UPSERT,
-            entityUrn=database_urn,
-            aspect=database_properties,
-        )
-        self.graph.emit(metadata_event)
-        logger.info(f"Properties updated for Database {name} ")
+        raise EntityDoesNotExist(f"Chart with urn: {urn} does not exist")
 
-        # set platform to athena
-        metadata_event = MetadataChangeProposalWrapper(
-            entityUrn=database_urn,
-            aspect=DataPlatformInstance(
-                platform="urn:li:dataPlatform:athena",
-            ),
-        )
-        self.graph.emit(metadata_event)
-        logger.info(f"Platform updated for Database {name} ")
+    # to expand on and replace `list_database_tables` will need new graphql query i expect
+    # but will be more equivelent to the get chart and table details methods.
+    def get_database_details(self, urn: str):
+        pass
 
-        # container type update
-        metadata_event = MetadataChangeProposalWrapper(
-            entityUrn=database_urn,
-            aspect=SubTypesClass(typeNames=[DatasetContainerSubTypes.DATABASE]),
-        )
-        self.graph.emit(metadata_event)
-        logger.info(f"Type updated for Database {name} ")
+    def upsert_table(self, metadata: Table):
+        if RelationshipType.PARENT not in metadata.relationships:
+            raise ValueError("A parent entity needs to be included in relationships")
 
-        return database_urn
-
-    def upsert_athena_table(
-        self,
-        metadata: TableMetadata,
-        database_metadata: DatabaseMetadata | None = None,
-    ) -> str:
-        if not metadata.parent_entity_name:
-            raise ValueError("parent_entity_name needs to be set in TableMetadata")
-
-        fully_qualified_name = f"{metadata.parent_entity_name}.{metadata.name}"
+        parent_name = metadata.relationships[RelationshipType.PARENT][0].display_name
+        fully_qualified_name = f"{parent_name}.{metadata.name}"
```

Comment:
TODO: I think it's probably worth extracting out anything that constructs names from other names - need to make sure this is consistent